src/main/java/service/operation/SupplyOperation.java

```diff
@@ -0,0 +1,21 @@
+package service.operation;
+
+import data.db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void operation(FruitTransaction.Operation operation, FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getQuantity() <= 0) {
```

Comment:
Performing the quantity check is good, but instead of silently returning, consider throwing an IllegalArgumentException to inform the caller that the quantity provided is invalid.